src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -119,7 +119,23 @@ protected override TaintedDataAbstractValue GetDefaultValueForParameterOnEntry(I
             {
                 if (this.DataFlowAnalysisContext.SourceInfos.IsSourceParameter(parameter, WellKnownTypeProvider))
                 {
-                    return TaintedDataAbstractValue.CreateTainted(parameter, parameter.DeclaringSyntaxReferences[0].GetSyntax(), this.OwningSymbol);
+                    // Location of the parameter, so we can track where the tainted data appears in code.
+                    SyntaxNode parameterSyntaxNode;
+                    if (!parameter.DeclaringSyntaxReferences.IsEmpty)
+                    {
+                        parameterSyntaxNode = parameter.DeclaringSyntaxReferences[0].GetSyntax();
+                    }
+                    else if (!parameter.ContainingSymbol.DeclaringSyntaxReferences.IsEmpty)
```

Comment:
Could we add a comment with an example for this case so that later on when reviewing we easily understand.

Also, you said you would want to ignore the case but as far as I understand this case is going to consider the property as tainted (I mean at least as tracked), isn't it?

Reply:
Ah ok, it is excluded with the change in `WebInputSources` but you still want robustness here in case we have another similar case. 

Depending on people there seems to be different approach but I have seen a couple of comments on roslyn going in the direction of let's fail if that's an unexpected scenario so that we can figure them out and do the right fix/change. So do we want to keep failures to discover other cases where we don't have the syntax?

Reply:
Sure, added comments

For this case, I think I prefer robustness (and if anyone uses debug builds, debug asserts will show up). Worst case is a false negative.